src/corecel/cont/Span.hh

```diff
@@ -33,22 +35,30 @@ constexpr std::size_t dynamic_extent = detail::dynamic_extent;
  *
  * Notably, only a subset of the functions (those having to do with size) are
  * \c constexpr. This is to allow debug assertions.
+ *
+ * Span can be instantiated with the special marker type \c LdgValue<T> to
+ * optimize reading constant data on device memory. In that case, data returned
+ * by \c front, \c back, \c operator[] and \c begin / \c end iterator use value
+ * semantics instead of reference. \c data still returns a pointer to the data
+ * and can be used to bypass using \c LdgIterator
  */
 template<class T, std::size_t Extent = dynamic_extent>
 class Span
 {
+    using SpanTraitsT = detail::SpanTraits<T>;
+
   public:
     //!@{
     //! \name Type aliases
-    using element_type = T;
-    using value_type = std::remove_cv_t<T>;
+    using element_type = typename SpanTraitsT::element_type;
+    using value_type = std::remove_cv_t<element_type>;
```

Comment:
I thought about this more after you pointed out that `const value_type` is ok even when `value_type` is const. The standard implementation of `std::array` and `std::span` both literally alias `value_type = T`, so maybe we should keep `value_type` const but let `element_type` be the non-const version?

Reply:
Looking at [std::span](https://en.cppreference.com/w/cpp/container/span), `element_type = T` and `value_type` removes cv qualifiers which is what we have now so I think it's fine. `std::array` is a bit different since it only has `value_type`

Reply:
Aha quite right. Thanks!